kusto/ingest/internal/queued/queued.go

```diff
@@ -264,27 +269,29 @@ func CompleteFormatFromFileName(props *properties.All, from string) error {
 }
 
 // upstreamContainer randomly selects a container queue in which to upload our file to blobstore.
-func (i *Ingestion) upstreamContainer() (azblob.ContainerURL, error) {
+func (i *Ingestion) upstreamContainer() (azblob.ContainerClient, error) {
 	mgrResources, err := i.mgr.Resources()
 	if err != nil {
-		return azblob.ContainerURL{}, errors.E(errors.OpFileIngest, errors.KBlobstore, err)
+		return azblob.ContainerClient{}, errors.E(errors.OpFileIngest, errors.KBlobstore, err)
 	}
 
 	if len(mgrResources.Containers) == 0 {
-		return azblob.ContainerURL{}, errors.ES(
+		return azblob.ContainerClient{}, errors.ES(
 			errors.OpFileIngest,
 			errors.KBlobstore,
 			"no Blob Storage container resources are defined, there is no container to upload to",
 		).SetNoRetry()
 	}
 
 	storageURI := mgrResources.Containers[rand.Intn(len(mgrResources.Containers))]
-	service, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net?%s", storageURI.Account(), storageURI.SAS().Encode()))
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net?%s", storageURI.Account(), storageURI.SAS().Encode())
```

Comment:
Does this need any URL escaping?

Reply:
I don't think so, we don't accept that data from the user - we accept it straight from kusto, and we do escape the SAS (with .Encode)